build.gradle

```diff
@@ -76,4 +76,78 @@ project(':configurator') {
                     'Main-Class': mainClassName)
         }
     }
-}
\ No newline at end of file
+}
+ext {
+    systemNames = ["Linux","Windows","Mac"]
+    jreDir = new File("$buildDir/jre.tar.gz")
+    jarDir = new File("$buildDir/libs")
+}
+
+import java.security.MessageDigest
+task downloadJre  {
+    def originalFileHash = "dd28d6d2cde2b931caf94ac2422a2ad082ea62f0beee3bf7057317c53093de93"
+    // Download JRE
+    doFirst {
+        if(!jreDir.exists()) {
+            def downloadUrl = new URL("https://github.com/AdoptOpenJDK/openjdk8-binaries/releases/download/jdk8u212-b03/OpenJDK8U-jdk_x64_linux_hotspot_8u212b03.tar.gz")
+            def file = new FileOutputStream(jreDir)
+            def out = new BufferedOutputStream(file)
+            try{
+                out << downloadUrl.openStream()
+                out.close()
+            }
+            catch(IOException e){
+                e.printStackTrace()
+            }
+            finally {
+                file.close()
+            }
+        }
+    }
```

Comment:
How's about you rework this `doFirst` to make it a bit more groovy:

```groovy
       if(!jreDir.exists()) {
            jreDir.mkdirs()
        }

        final filename = 'OpenJDK8U-jdk_x64_linux_hotspot_8u212b03.tar.gz'
        new URL("https://github.com/AdoptOpenJDK/openjdk8-binaries/releases/download/jdk8u212-b03/$filename")
                .withInputStream {
                    new File( "${jreDir.getAbsolutePath()}/${filename}" ) << it
                }
```

Note how `filename` has been used. You'll obviously need to externalise this to use in your packaging task.